### PR TITLE
Implement dark mode with CSS invert filter

### PR DESCRIPTION
**Pull request by axelboc**
image

image

### CODE REVIEW

File: src/h5web/vis-packs/core/heatmap/ColorBar.module.css [hunk 1]
```diff
@@ -21,9 +21,15 @@
...
 
-.gradient {
+.gradientBar {
   grid-area: gradient;
+  display: flex;
   border: 1px solid var(--black);
+}
+
+.gradient {
+  composes: no-invert from global;
```
Comment (axelboc) [inline src/h5web/vis-packs/core/heatmap/ColorBar.module.css:31]: I had move the gradient to a child element so as to not invert the border of the gradient bar.

File: src/h5web/vis-packs/core/shared/VisCanvas.module.css [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 /* R3F's root element */
 .canvasWrapper {
+  composes: no-invert-canvas from global;
```
Comment (axelboc) [inline src/h5web/vis-packs/core/shared/VisCanvas.module.css:16]: R3F doesn't provide a way to add a class to the canvas, and composing a class from an element selector like `.canvasWrapper > canvas { composes: no-invert from global; }` is technically not possible.

File: src/styles/base.css [hunk 3]
```diff
@@ -62,7 +62,27 @@ abbr[title] {
   *,
   *::before,
   *::after {
-    transition: none !important;
-    animation: none !important;
+    animation-duration: 0.01ms !important;
+    animation-iteration-count: 1 !important;
+    transition-duration: 0.01ms !important;
```
Comment (axelboc) [inline src/styles/base.css:67]: This reset code has been going around. It avoids breaking some types of transition/animation code. Don't think it matters for us, but I thought we might as well be safe.

**Comment by axelboc** [comment]
Ah hang on, I'll add an E2E test.

**Comment by axelboc** [comment]
Hmm, couldn't find an easy way to tell Cypress to switch to dark mode... Oh well, not a big deal.

File: src/styles/index.css [hunk 4]
```diff
@@ -1,3 +1,2 @@
 @import 'vars.css';
 @import 'base.css';
-@import 'utils.css';
```
Comment (axelboc) [inline src/styles/index.css:3]: I've merged the content of `utils.css` into `base.css`. We had reflex overrides in there that weren't `utils` anyway, and in the dark mode query, I use a couple of utility classes, so it doesn't seem worth it to try to have two separate files.

**Review by loichuder — APPROVED** [review]
:sparkles: :new_moon_with_face:

**Pull request merged**